flink-runtime/src/main/java/org/apache/flink/runtime/state/ArrayListSerializer.java

```diff
@@ -53,11 +53,7 @@ public boolean isImmutableType() {
 
 	@Override
 	public ArrayList<T> copy(ArrayList<T> from) {
-		ArrayList<T> newList = new ArrayList<>(from.size());
-		for (int i = 0; i < from.size(); i++) {
-			newList.add(elementSerializer.copy(from.get(i)));
-		}
-		return newList;
+		return new ArrayList<>(from);
```

Comment:
Yes, you are absolutely right. I somehow oversaw the inner call to copy :D. Will change this back.